Sources/CryptoSwift/HashProtocol.swift

```diff
@@ -16,6 +16,10 @@ internal protocol HashProtocol: class {
 extension HashProtocol {
     
     func prepare(_ len:Int) -> Array<UInt8> {
+        return self.prepare(len, allowance: 8)
+    }
+    
+    func prepare(_ len:Int, allowance: Int) -> Array<UInt8> {
```

Comment:
I'm not sure. There may be some instances where one might want a 0 (mod 128) padding but with only an 8-byte integer representation of the message length - making the allowance to be only 8 rather than 128 / 8 or 16.

If `len / 8` is mostly used, we can set the default return of `prepare(_ len: Int) -> Array<UInt8>` to `return self.prepare(len, allowance: len / 8)` and retain the `prepare(_:allowance:)` function to have a function for customizing the allowance.